File: python/prime-factors/prime_factors.py
def prime_factors(number):
    factors = []

    def append_factor(fact):
        nonlocal number
        while number % fact == 0:
            factors.append(fact)
            number /= fact

    append_factor(2)
    append_factor(3)

    possible_prime = 5
    add2 = True
    while number > 1:
        append_factor(possible_prime)

        possible_prime += 2 if add2 else 4
        add2 = not add2

    return factors

File: lua/prime-factors/prime-factors.lua
local prime_numbers = { 2, 3, 5, 11, 17, 23, 461, 9539, 894119 }

local function is_divisible(dividend, divisor)
    return dividend % divisor == 0
end

local function prime_factors(number)
    local factors = {}

    for _, prime in ipairs(prime_numbers) do
        while is_divisible(number, prime) do
            table.insert(factors, prime)
            number = number // prime
        end
    end

    return factors
end

return prime_factors
